src/django/pfb_analysis/tasks.py

```diff
@@ -23,5 +44,86 @@ def create_batch_from_remote_shapefile(shapefile_url):
+                directory=task.job.s3_results_path))
+
+            # Store the neigborhood ways and Census block results back to models
+            logging.info('Importing neighborhood ways and Census blocks back to database')
+            add_results_geoms(task.job)
+
+            # Mark this upload task and its associated analysis job as completed.
+            task.job.status = AnalysisJob.Status.COMPLETE
+            task.job.save()
+            task.status = AnalysisLocalUploadTask.Status.COMPLETE
+            task.save()
+            logging.info('Successfully completed upload task {uuid}'.format(
+                uuid=local_upload_task_uuid))
+        except Exception as ex:
+            logging.error('Failed to upload analysis results for task {uuid}'.format(
+                uuid=local_upload_task_uuid))
+            raise LocalAnalysisFetchException(ex.message)
+
+    def download_and_extract_local_results(tmpdir, task):
+        try:
+            local_filename = os.path.join(tmpdir, 'analysis_results.zip')
+            download_file(task.upload_results_url, local_filename=local_filename)
+            with zipfile.ZipFile(local_filename, 'r') as zip_handle:
+                zip_handle.extractall(tmpdir)
+            results_files = [filename for filename in os.listdir(tmpdir)]
+
+            # Verify all expected results files are in the upload
+            missing = LOCAL_ANALYSIS_FILES.difference(set(results_files))
+            if missing:
+                raise LocalAnalysisFetchException('Missing expected results files: {files}'.format(
+                    files=', '.join(missing)))
+        except Exception as ex:
+            msg = 'Failed to fetch analysis results for task {uuid} from {url}: {msg}'.format(
+                uuid=local_upload_task_uuid, url=task.upload_results_url, msg=ex.message)
+            logging.error(msg)
+            raise LocalAnalysisFetchException(ex.message)
+
+    try:
+        logging.info('Starting local analysis upload task {uuid}'.format(
+            uuid=local_upload_task_uuid))
+        tmpdir = tempfile.mkdtemp()
+
+        # First find the upload task to process and mark its status
+        task = AnalysisLocalUploadTask.objects.get(uuid=local_upload_task_uuid)
+        task.status = AnalysisLocalUploadTask.Status.IMPORTING
+        task.save()
+
+        # Run the local analysis download, extract, and upload for the task
+        logging.info('Starting local analysis upload {uuid} from {url}'.format(
+                     uuid=local_upload_task_uuid, url=task.upload_results_url))
+        download_and_extract_local_results(tmpdir, task)
+        upload_and_insert_local_results(tmpdir, task)
+    except (ObjectDoesNotExist, ValidationError):
+        logging.error('No local upload analysis task found for {uuid}.'.format(
+                      uuid=local_upload_task_uuid))
+    except LocalAnalysisFetchException as ex:
+        task.status = AnalysisLocalUploadTask.Status.ERROR
```

Comment:
I haven't tracked down why, but when I provide an invalid zip file, I see the "Missing expected results files" error in the log output, but the task status isn't getting changed from `CREATED`.

Reply:
The initial `POST` response will say `CREATED`. Does a subsequent requery still not show an error status? If so, could you share the URL to the test file?

Reply:
Doh, sorry.  I was looking at the `analysis_job` list, not the `local_upload_tasks` list.  The upload task is getting the proper error status and message.